### PR TITLE
[PVR][Estuary] Recordings Playlist Actions

### PR DESCRIPTION
**Pull request by ksooo**
This PR enhances recording folders and recordings with some play/playlist actions. Those actions are already available for normal videos and video folders and with this PR PVR catches up on this.

* Recording folders now have 'Play', 'Play next' and 'Queue item' context menu entries.
* Recordings now have Play next' and 'Queue item' context menu entries.
* PVR Recordings window side blade now has a button 'Go to playlist'

screenshot000
screenshot001
screenshot002

All changes are runtime-tested on macOS and Android, latest Kodi master.

@phunkyfish when you find some time for a code review.
@ronie could you take a look at the skin changes?

### CODE REVIEW

File: xbmc/video/ContextMenus.cpp [hunk 1]
```diff
@@ -119,7 +120,58 @@ bool CResume::IsVisible(const CFileItem& itemIn) const
...
+
+  // Determine the proper list to queue this element
+  int playlist = player.GetCurrentPlaylist();
+  if (playlist == PLAYLIST_NONE)
+    playlist = g_application.GetAppPlayer().GetPreferredPlaylist();
+  if (playlist == PLAYLIST_NONE)
+    playlist = PLAYLIST_VIDEO;
+
+  if (bPlayNext && g_application.GetAppPlayer().IsPlaying())
+    player.Insert(playlist, queuedItems, player.GetCurrentSong() + 1);
```
Comment (phunkyfish) [inline xbmc/video/ContextMenus.cpp:167]: This line feels wrong ;)

File: xbmc/video/ContextMenus.cpp [hunk 2]
```diff
@@ -164,7 +237,12 @@ bool CPlay::IsVisible(const CFileItem& itemIn) const
     return false;
 
   if (item.m_bIsFolder)
-    return false; //! @todo implement
+  {
+    if (StringUtils::StartsWith(item.GetPath(), "pvr://recordings/"))
+      return true;
+    else
+      return false; //! @todo implement
```
Comment (phunkyfish) [inline xbmc/video/ContextMenus.cpp:244]: What should be implemented here?

**Review by phunkyfish — COMMENTED** [review]
Just some comments. I'm interested in what still needs to be implemented.

File: xbmc/video/ContextMenus.cpp (hunk 2 above)
Comment (ksooo) [reply xbmc/video/ContextMenus.cpp:244]: Once video and music components are changing to the "new" context menu system, the respective implementation (which are now located in the resp. window classes) need to go here.

File: xbmc/video/ContextMenus.cpp (hunk 1 above)
Comment (ksooo) [reply xbmc/video/ContextMenus.cpp:167]: Nope, that's correct. The item shall be queued right after the currently playing song, thus '+ 1'. I copied that code from a video window class' implementation for that function.

**Review by ronie — APPROVED** [review]
skin changes look good.

File: xbmc/video/ContextMenus.cpp (hunk 1 above)
Comment (phunkyfish) [reply xbmc/video/ContextMenus.cpp:167]: Cool

File: xbmc/video/ContextMenus.cpp (hunk 2 above)
Comment (phunkyfish) [reply xbmc/video/ContextMenus.cpp:244]: Ah, ok. Makes sense.

**Review by phunkyfish — APPROVED** [review]

**Comment by emveepee** [comment]
How does a folder queue ensure that episodes get added in the proper logical viewing order since Kodi PVR doesn't support season episode sorting?

**Comment by ksooo** [comment]
@emveepee you answered your question yourself.

**Comment by ksooo** [comment]
This needs a bit more love:
* Bug in side blade navigation
* Sorting for folder play according to recordings window sort method

**Comment by ronie** [comment]
>     * Bug in side blade navigation

change Includes_MediaMenu.xml, line 222 to:
`<onup>302</onup>`

**Comment by ksooo** [comment]
@ronie thanks.

**Comment by ksooo** [comment]
@ronie  @phunkyfish ready for re-review

**Review by ronie — APPROVED** [review]

**Review by phunkyfish — APPROVED** [review]
LGTM

**Pull request merged**